stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/HealthMonitor.kt

```diff
@@ -16,94 +16,150 @@
 
 package io.getstream.chat.android.client.socket
 
-import android.os.Handler
-import android.os.Looper
+import io.getstream.chat.android.client.utils.TimeProvider
 import io.getstream.logging.StreamLog
-import java.util.Date
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import kotlin.math.floor
 import kotlin.math.max
 import kotlin.math.min
 
-private const val HEALTH_CHECK_INTERVAL = 10 * 1000L
-private const val MONITOR_INTERVAL = 1000L
-private const val NO_EVENT_INTERVAL_THRESHOLD = 30 * 1000L
-private const val MONITOR_START_DELAY = 1000L
+private const val HEALTH_CHECK_INTERVAL = 1_000L
+private const val MONITOR_INTERVAL = 10_000L
+private const val NO_EVENT_INTERVAL_THRESHOLD = 30_000L
 
-internal class HealthMonitor(private val checkCallback: () -> Unit, private val reconnectCallback: () -> Unit) {
+internal class HealthMonitor(
+    private val timeProvider: TimeProvider = TimeProvider,
+    private val retryInterval: RetryInterval = ExponencialRetryInterval,
+    private val coroutineScope: CoroutineScope,
+    private val checkCallback: () -> Unit,
+    private val reconnectCallback: () -> Unit,
```

Comment:
Not sure if that's related to this PR but I managed to get permanent disconnected state by just waiting on the channels screen. It ties to reconnect and the `reconnectCallback` callback is invoked. However this distinct check is never passed:

```
internal fun List<LifecyclePublisher>.combine(): Flow<Event.Lifecycle> {
    return combine(this.map { it.lifecycleEvents }) {
        it.toList().combineLifecycleEvents()
    }
        .distinctUntilChanged { old, new -> old == new || (old.isStopped() && new.isStopped()) }
}
```

Reply:
The old and new states are "Started" even though we are disconnected